Sources/BinaryParsing/Parsers/InlineArray.swift

```diff
@@ -27,6 +27,21 @@ extension InlineArray where Element == UInt8 {
   }
 }
 
+@available(macOS 26, iOS 26, watchOS 26, tvOS 26, visionOS 26, *)
+extension InlineArray: @retroactive Equatable where Element == UInt8 {
+  @inlinable
+  public static func == (
+    lhs: InlineArray<count, UInt8>, rhs: InlineArray<count, UInt8>
+  ) -> Bool {
+    for i in 0..<count {
+      if lhs[i] != rhs[i] {
+        return false
+      }
+    }
+    return true
+  }
+}
+
```

Comment:
We don't want to add this conformance here, as it can cause problems if `InlineArray` actually gains its own `Equatable` conformance in the future. Let's just have an internal `isEqual(to:)` method to handle this kind of comparison.